src/libs/actions/Task.ts

```diff
@@ -703,19 +719,30 @@ function getShareDestination(reportID: string, reports: OnyxCollection<OnyxTypes
     };
 }
 
+/**
+ * Returns the parentReportAction if the given report is a thread/task.
+ *
+ * @deprecated Use Onyx.connect() or withOnyx() instead
+ */
+function getParentReportAction(report: OnyxEntry<OnyxTypes.Report>): ReportAction | Record<string, never> {
```

Comment:
Would it be possible and make sense to pass `Report` here and to `deleteTask` instead of `OnyxEntry<OnyxTypes.Report>`, so that we don't have to check `report?`? It looks like we only call deleteTask in one place

It looks like before we were using `props.report.reportName` and not `props.report?.reportName`

Reply:
I could revert the changes in `deleteTask` to go back to using `const taskReport = ReportUtils.getReport(taskReportID);` which returns a `Report` and not a `OnyxEntry<OnyxTypes.Report>`, but here is why I removed it.

- The view already has the full report object from Onyx
- The view was passing 4 properties of the report object to `deleteTask` and then getting the full report object from `ReportUtils.getReport()`
- This is inefficient and leads to really bad code and possibly stale data
- By using the same report reference in all the methods, it's assured that the report object will always be correct

Reply:
Oh yeah I think the change you made was an improvement, so I wasn't suggesting reverting back to the old version, just to go from `getParentReportAction(report: OnyxEntry<OnyxTypes.Report>):` to `getParentReportAction(report: Report):`, and remove a lot of `?` that would be unnecessary. Is that doable?

Reply:
You can't simply cast one type to another like that, but [I asked about it in Slack](https://expensify.slack.com/archives/C0593T50UHG/p1709232447912569) and I at least found a way to remove all the optional chaining by adding an early return at the beginning of the method.

Reply:
That works. I was thinking not just updating the param type --that was to illustrate what I meant, but also the function call `Session.checkIfActionIsAllowed(Task.deleteTask(props.report));`, with something like `Session.checkIfActionIsAllowed(Task.deleteTask(props.report as Report));` or whatever TS magic helps with that.

I like checking before and returning early tho 👍